src/runtime/task/labels.rs

```diff
@@ -0,0 +1,330 @@
+/*
+** Code directly copied from https://github.com/hyperium/http/blob/master/src/extensions.rs
+** but renaming 'Extensions' to 'Labels'
+*/
```

Comment:
Would be nice with a doc comment explaining what the code does.

Reply:
Added.

Reply:
nit: `//!`